src/AnimationRunners.ts

```diff
@@ -248,3 +250,53 @@ export const loop = (loopConfig: LoopProps) => {
     state.position
   ]);
 };
+
+export const toggle = (params: {
+  clock?: Animated.Clock;
+  toggleState: Animated.Value<number>;
+  duration?: number;
+  from: Animated.Adaptable<number>;
+  to: Animated.Adaptable<number>;
+  easing?: Animated.EasingFunction;
+  value: Animated.Value<number>;
+}) => {
+  const { toggleState, easing, duration, value, to, from, clock } = {
+    clock: new Clock(),
+    duration: 250,
+    easing: Easing.linear,
+    from: new Value(0),
+    toggleState: new Value(GestureState.END),
+    to: new Value(1),
+    value: new Value(0),
+    ...params
+  };
+
+  return [
+    cond(eq(toggleState, 1), [
+      set(
+        value,
+        timing({
+          clock,
+          duration,
+          easing,
+          from: value,
+          to
+        })
+      ),
+      cond(not(clockRunning(clock)), [set(toggleState, GestureState.END)])
```

Comment:
@dsznajder is GestureState.END involved here?

Reply:
I just used it as `toggleState` when nothing should happen.

Reply:
Why should this function have a dependency with `GestureState`?

Reply:
@wcandillon I just use it as some descriptive value. I can use just some number (i.e. -1). What do you think?